lion/io/database_parameters.h

```diff
@@ -50,39 +50,51 @@ inline void read_parameters(Xml_document& doc, const std::string& path, const st
 {
     assert(p.size() == used_parameters.size());
 
+    Xml_element root = doc.get_root_element();
+    Xml_element element = doc.get_element(path.substr(0, path.find_last_of("/")));
```

Comment:
Why not `Xml_element element =doc.get_element(path + p[i].name)` ?

Reply:
Sorry I don't quite get your question

Reply:
I was just thinking of the following case:

`path = "a/b"`
`parameter = "c/d"`

the algorithm as it is it would only mark `a`, `b`, and `d`, but not `c`. Right?

Reply:
Yup, true I did not think about nested parameters. I'll change this as well

Reply:
Yup, I now changed it to be inside the for loop. This makes sure it won't happen.

Reply:
I would just do the recursive search inside the parameters loop. Once you reach the first ancestor that has got the __unused__ attribute, you can stop there and continue to the next parameter.

Another stupid thing, can you change `__unused__="false"` to `__used__="true"`? It now looks very weird to me the other way.. my fault.